src/pages/AddCertificate/AddCertificate.tsx

```diff
@@ -0,0 +1,248 @@
+
+    return isValid;
+  };
+
+  const handleInput = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { name, value } = e.target;
+    setCertificateData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const handleDateChange = (name: string, date: Date | null) => {
+    setCertificateData((prevData) => ({
+      ...prevData,
+      [name]: date,
+    }));
+  };
+
+  const handleSave = () => {
+    console.log('Attempting to save...');
+    if (validateForm()) {
+      console.log('Form validated successfully');
+      setCertificateData(certificateData);
+      certificates.push(certificateData);
+      console.log('Certificate added:', certificateData);
+      console.log('Updated certificates array:', certificates);
+
+      setErrors({
+        supplier: '',
+        certificateType: '',
+        validFrom: '',
+        validTo: '',
+        validRange: '',
+      });
+
+      navigate('/machine-learning/example1');
+    } else {
+      console.log('Form validation failed');
+      console.log('Current errors:', errors);
+    }
+  };
+
+  const handleReset = () => {
+    setCertificateData({
+      supplier: '',
+      certificateType: '',
+      validFrom: null,
+      validTo: null,
+    });
+    setPdfPreview(null);
+  };
+
+  const handlePdfUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = () => {
+        setPdfPreview(reader.result as string);
+      };
+      reader.readAsDataURL(file);
+    }
+  };
+
+  return (
+    <div className="certificate-form-container">
+      <div className="left-side">
+        <div className="row">
+          <label htmlFor="supplier">Supplier</label>
+          <div className="input-wrapper">
+            <input
+              type="text"
+              id="supplier"
+              name="supplier"
+              value={certificateData.supplier}
+              onChange={handleInput}
+              className="supplier-input"
+            />
+
+            <button className="search-button">
+              <SearchIcon />
+            </button>
+
+            <button className="clear-button">
+              <CrossIcon />
+            </button>
+          </div>
+        </div>
+        {errors.supplier && <p className="error-message">{errors.supplier}</p>}
+
+        <div className="row">
+          <label htmlFor="certificate-type">Certificate type</label>
+          <select
+            id="certificate-type"
+            name="certificateType"
+            value={certificateData.certificateType}
+            onChange={handleInput}
+            defaultValue=""
+          >
+            <option
+              value=""
+              disabled
+            >
+              Select your option
+            </option>
+            <option value="Permission of Printing">
+              Permission of Printing
```

Comment:
Create enum for certificate type please, and then create array out of enum values and pass into your generic dropdown component

Reply:
not resolved yet